src/db.c

```diff
@@ -294,6 +295,7 @@ static const struct col_type_map qi_cols_map[] =
     { "bitrate",            qi_offsetof(bitrate),             DB_TYPE_INT },
     { "samplerate",         qi_offsetof(samplerate),          DB_TYPE_INT },
     { "channels",           qi_offsetof(channels),            DB_TYPE_INT },
+    { "flag",               qi_offsetof(flag),                DB_TYPE_INT },
```

Comment:
Why add it to queue_item? I suppose this mark is very much like rating, which isn't part of queue_item

Reply:
the playing track would have corresponding mark - this allows the user (if/when we add to the UI) not to see what the status of the current mark is without a need for join on `files` tbl.

Reply:
Please explain why it is different than rating

Reply:
Are you suggesting that `usermark` be derived at the point that `now playing` calling `jsonapi_reply_player()` perform a join to get `usermark`?

Or are you questioning why `usermark` is exposed for items in the queue?

For first, we are getting items like `channels`, `samplerate`, in the same manner thus `usermark` is modelled on that.  Regading `rating`, this is not currently exposed on the queue or now playing

For second, without `usermark` being exposed how would a user determine the current `usermark` value?  For the use case of "i hate this song" or "this song was encoded wrong", you'd expect this to done whilst the user is listening or just listened to that track, otherwise you'd have to go back into the the underlying album listing to then mark the track - this would largely be a negative UX

Reply:
> For first, we are getting items like channels, samplerate, in the same manner thus usermark is modelled on that. 

What separates rating and usermark from channels and samplerate is that they only make sense for library items, not for one-time queue items.

> For second, without usermark being exposed how would a user determine the current usermark value? For the use case of "i hate this song" or "this song was encoded wrong", you'd expect this to done whilst the user is listening or just listened to that track, otherwise you'd have to go back into the the underlying album listing to then mark the track - this would largely be a negative UX

Completely agree. Since rating is also a way saying "I hate this song", I was assuming that the web UI currently had a way of getting and setting rating during playback, and thus that there already was a solution for this that you could lean on without adding field duplication. Like how it is done when the Remote app is controller. However, it seems I am mistaken, and that this isn't actually supported.

Wouldn't it be possible to get by getting the file id from the queue, then retrieving the usermark value with GET /api/library/tracks/{id}? And then setting the value with PUT /api/library/tracks/{id}?

BTW, in any case you need to documentation of usermark to object descriptions in README_JSON_API.md

Reply:
> Wouldn't it be possible to get by getting the file id from the queue, then retrieving the usermark value with GET /api/library/tracks/{id}? And then setting the value with PUT /api/library/tracks/{id}?

This is possible - the webui has a `queue` in its `store` which includes the `track_id` so we this can be requested via the _queue position id_ which is what the `now playing` page has.

Depends on the implemetnation, but if we go with the ability to mark the track via the modal dialog, then we'd only ask for the full track info (via GET `/api/library/tracks/{id}`) when user clicks for that.

Will drop from `queue` table based on this

Reply:
not directly related to this PR but i've prototyped the get and set for the web ui using the `ModalDialogQueueItem` (avilable on the queue listing and now playing pages) and it works with the requests back to the server via the get/set `/api/library/tracks/{id}`

will submit that as seperate PR once this is one is done.